File: src/main/java/com/aries/tutorial/util/EventFormatter.java
package com.aries.tutorial.util;

import com.aries.extension.data.EventData;

public final class EventFormatter {

    private EventFormatter() {
    }

    public static String format(int idx, EventData data) {
        StringBuilder sb = new StringBuilder(192);
        sb.append("▸ #").append(idx).append("  ");
        sb.append('⟨').append(data.eventLevel).append("⟩  ");
        sb.append("domain#").append(data.domainId).append('·').append(data.instanceName);
        sb.append("  │  biz=").append(data.businessName);
        sb.append("  │  tx=").append(data.txid);
        sb.append("  │  svc=").append(data.serviceName);
        sb.append("  │  ").append(data.errorType);
        return sb.toString();
    }
}

File: src/main/kotlin/com/aries/tutorial2/util/EventFormatter.kt
package com.aries.tutorial2.util

import com.aries.extension.data.EventData

object EventFormatter {

    fun format(idx: Int, data: EventData): String = buildString(192) {
        append("▸ #").append(idx).append("  ")
        append('⟨').append(data.eventLevel).append("⟩  ")
        append("domain#").append(data.domainId).append('·').append(data.instanceName)
        append("  │  biz=").append(data.businessName)
        append("  │  tx=").append(data.txid)
        append("  │  svc=").append(data.serviceName)
        append("  │  ").append(data.errorType)
    }
}
